lib/sidekiq_publisher/compatibility.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+module SidekiqPublisher
+  module Compatibility
+    class << self
+      # Sidekiq::Worker will be renamed to Sidekiq::Job in sidekiq 7.0.0 and a
+      # deprecation warning will be printed in sidekiq 6.4.0, per mperham/sidekiq#4971.
+      # Sidekiq 6.2.2 (mperham/sidekiq@8e36432) introduces an alias which is used here
+      # to ensure future compatibility.
+      # TODO: Update this comment and the version constraint below once a new version of
+      # sidekiq is released which automatically requires sidekiq/job
+      def sidekiq_job_class
+        @_sidekiq_job_class ||= Gem::Dependency.new("sidekiq", "> 6.2.2").then do |dependency|
```

Comment:
TODO: Once a new version of sidekiq is released which [pulls in `sidekiq/job`](mperham/sidekiq@86493a01cab005c73c32477039d0ff6eae1ecb28), update the constraint.